src/App.tsx

```diff
@@ -1,61 +1,143 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { useState } from 'react';
+
+export type User = {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+};
+
+export type RawTodo = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+};
+
+export type TodoWithUser = RawTodo & {
+  user?: User;
+};
+
+function getUserById(users: User[], userId: number): User | undefined {
+  return users.find(user => user.id === userId);
+}
+
+export const initialTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(usersFromServer, todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState('');
+  const [currentTodos, setCurrentTodos] =
+    useState<TodoWithUser[]>(initialTodos);
+
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleReset = () => {
+    setTitle('');
+    setSelectedUserId('');
+    setTitleError(false);
+    setUserError(false);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    if (titleError) {
+      setTitleError(false);
+    }
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(event.target.value);
+    if (userError) {
+      setUserError(false);
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    let hasError = false;
+
+    if (title.trim() === '') {
+      setTitleError(true);
+      hasError = true;
+    }
+
+    if (selectedUserId === '') {
+      setUserError(true);
+      hasError = true;
+    }
```

Comment:
if you just want to return if `hasError` is true, then just return from `if`'s


```suggestion
    if (title.trim() === '') {
      setTitleError(true);
     return;
    }

    if (selectedUserId === '') {
      setUserError(true);
      return;
    }
```